splinterd/tests/framework/network.rs

```diff
@@ -70,22 +75,31 @@ impl Network {
             ));
 
             self.temp_dirs.insert(node.node_id().to_string(), temp_dir);
-            self.nodes.push(node);
+            self.nodes.push(NetworkNode::Node(node));
         }
 
         for node in &self.nodes {
-            let registry_writer = node.registry_writer();
-            for (node_id, pub_key, endpoints) in &registry_info {
-                registry_writer
-                    .add_node(
-                        RegistryNode::builder(node_id)
-                            .with_display_name(node_id)
-                            .with_endpoints(endpoints.to_vec())
-                            .with_key(pub_key.as_hex())
-                            .build()
-                            .map_err(|e| InternalError::from_source(Box::new(e)))?,
-                    )
-                    .map_err(|e| InternalError::from_source(Box::new(e)))?;
+            match node {
+                NetworkNode::Node(node) => {
+                    let registry_writer = node.registry_writer();
+                    for (node_id, pub_key, endpoints) in &registry_info {
+                        registry_writer
+                            .add_node(
+                                RegistryNode::builder(node_id)
+                                    .with_display_name(node_id)
+                                    .with_endpoints(endpoints.to_vec())
+                                    .with_key(pub_key.as_hex())
+                                    .build()
+                                    .map_err(|e| InternalError::from_source(Box::new(e)))?,
+                            )
+                            .map_err(|e| InternalError::from_source(Box::new(e)))?;
+                    }
+                }
+                _ => {
+                    return Err(InternalError::with_message(
```

Comment:
This could probably be an `unreachable!()` stmt.

Currently, this method isn't written to support multiple calls correctly, as it iterates over the whole list of nodes.  We  should probably correct that in a future PR.

Reply:
Replaced with `unreachable!()`